src/utils/websockets.js

```diff
@@ -38,11 +41,28 @@ export function connectWS(topic, onMessageCallback) {
   };
 
   const connect = () => {
+    let reconnectCounter = 0;
     this.sockClient = new Client({
       brokerURL: config.WS_URL,
-      // debug: function(str) {
-      //   console.log('debug: ', str);
-      // },
+      debug(strMessage) {
+        // console.log('debug: ', strMessage);
+        // -- detect reconnect and increment the reconnect counter
+        if (strMessage.includes('reconnect')) {
+          getUserSession()
+            .then(({ data }) => {
+              reconnectCounter =
+                data && !data.loggedIn ? reconnectCounter + 1 : 0;
+            })
+            .catch(() => {
+              reconnectCounter += 1;
+            });
+        }
+        // -- if more than max allowed reconnect times  ->  deactivate
+        if (reconnectCounter > maxReconnectTimesNo) {
+          this.reconnectDelay = 0; // 0 - deactivates the sockClient
+          store.dispatch(noConnection(true));
```

Comment:
so we here are stopping the socket, but shouldn't we also redirect to the login page?

Reply:
That was my initial approach but for a nicer user experience user should see an error on the screen. He will be somehow forced to do the refresh seeing that message.